### ISSUE TITLE
MATLAB kernel issue - Dead Kernel

### ISSUE THREAD

**Issue by mittal4792** [ipython/ipython#9997]
Creating a new notebook with Python works fine, but creating one with Matlab does not work.
It shows the following error.
I believe I have installed the matlab_kernel properly.

Error trace:
[I 15:53:32.848 NotebookApp] [nb_conda_kernels] enabled, 3 kernels found
[I 15:53:33.430 NotebookApp] [nb_conda] enabled
[I 15:53:33.492 NotebookApp] [nb_anacondacloud] enabled
[I 15:53:33.540 NotebookApp] \u2713 nbpresent HTML export ENABLED
[W 15:53:33.543 NotebookApp] \u2717 nbpresent PDF export DISABLED: No module named 'nbbrowserpdf'
[I 15:53:33.667 NotebookApp] Serving notebooks from local directory: H:\
[I 15:53:33.669 NotebookApp] 0 active kernels
[I 15:53:33.671 NotebookApp] The Jupyter Notebook is running at: http://localhost:8880/
[I 15:53:33.674 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[I 15:53:47.130 NotebookApp] Kernel started: 465d2f61-6d0a-4eca-8272-3bd5efced378
[I 15:53:53.302 NotebookApp] Kernel started: aeb0f5b8-68a5-4702-8405-b3d52d5f39d5
C:\Users\mittalp\AppData\Local\Continuum\Anaconda3\python.exe: No module named matlab_kernel
[I 15:53:56.302 NotebookApp] KernelRestarter: restarting kernel (1/5)
C:\Users\mittalp\AppData\Local\Continuum\Anaconda3\python.exe: No module named matlab_kernel
[I 15:53:59.319 NotebookApp] KernelRestarter: restarting kernel (2/5)
C:\Users\mittalp\AppData\Local\Continuum\Anaconda3\python.exe: No module named matlab_kernel
[I 15:54:02.344 NotebookApp] KernelRestarter: restarting kernel (3/5)
C:\Users\mittalp\AppData\Local\Continuum\Anaconda3\python.exe: No module named matlab_kernel
[W 15:54:03.364 NotebookApp] Timeout waiting for kernel_info reply from aeb0f5b8-68a5-4702-8405-b3d52d5f39d5
[I 15:54:05.366 NotebookApp] KernelRestarter: restarting kernel (4/5)
WARNING:root:kernel aeb0f5b8-68a5-4702-8405-b3d52d5f39d5 restarted
C:\Users\mittalp\AppData\Local\Continuum\Anaconda3\python.exe: No module named matlab_kernel
[W 15:54:08.380 NotebookApp] KernelRestarter: restart failed
[W 15:54:08.383 NotebookApp] Kernel aeb0f5b8-68a5-4702-8405-b3d52d5f39d5 died, removing from map.
ERROR:root:kernel aeb0f5b8-68a5-4702-8405-b3d52d5f39d5 restarted failed!
[W 15:54:08.451 NotebookApp] Kernel deleted before session
[W 15:54:08.456 NotebookApp] 410 DELETE /api/sessions/48b4d31c-8951-4ba3-a618-baddaaff18e1 (::1) 6.51ms referer=http://localhost:8880/notebooks/Projects/Untitled%20Folder/Untitled1.ipynb

![1](https://cloud.githubusercontent.com/assets/18646711/19286285/16ff94f8-8fcb-11e6-8052-e6ba990c1f8b.JPG)

**Comment by takluyver** [comment]
Here's the important line from that:

```
C:\Users\mittalp\AppData\Local\Continuum\Anaconda3\python.exe: No module named matlab_kernel
```

It can't find an importable module called `matlab_kernel`.

**Comment by mittal4792** [comment]
@takluyver Precisely. That's the issue to be considered since I installed the matlab_kernel at the correct location, still I am facing this error.

**Comment by takluyver** [comment]
Can you, at a command prompt, run:

```
C:\Users\mittalp\AppData\Local\Continuum\Anaconda3\python.exe -c "import matlab_kernel"
```

How did you install it?

**Comment by mittal4792** [comment]
Running the suggested command, I got this:

> Traceback (most recent call last):
> File "\<string\>", line 1, in \<module\>
> ImportError: No module named 'matlab_kernel'

To install the kernel, the commands I used are:

> pip install matlab_kernel
> python -m matlab_kernel install

**Comment by takluyver** [comment]
What about `where python` and `where pip`?

**Comment by mittal4792** [comment]
> where python
> C:\Users\mittalp\AppData\Local\Continuum\Anaconda3\python.exe

> where pip
> C:\Users\mittalp\AppData\Local\Continuum\Anaconda3\Scripts\pip.exe

**Comment by dsblank** [comment]
@mittal4792 BYW, You're reporting this in the wrong issue tracker. This is the ipython kernel issues. The correct tracker is: https://github.com/Calysto/matlab_kernel/

Your problem is that you need to use the pip and python from your environment when installing components. That might be something like:

```
C:\Users\mittalp\AppData\Local\Continuum\Anaconda3\python.exe -m matlab_kernel install
```

**Comment by mittal4792** [comment]
@dsblank Thank you for informing me about the correct place to track the issue.

As for you suggested command, I am getting this : 
> C:\Users\mittalp\AppData\Local\Continuum\Anaconda3\python.exe: No module named matlab_kernel

**Comment by StanczakDominik** [comment]
Given the lack of activity on this and the fact that this is not an ipython issue... Close, maybe? :)
